Misc/NEWS.d/next/C_API/2024-08-12-10-15-19.gh-issue-109523.S2c3fi.rst

```diff
@@ -0,0 +1 @@
+In _io_TextIOWrapper_read_impl, raise BlockingIOError when read() from a non-blocking pipe does not return bytes.
```

Comment:
This updates `_pyio` as well now, Maybe something like "Reading text from a non-blocking stream with `.read()` may now throw a NonBlockingIO error when Non-blocking I/O is encountered."

Reply:
It took a bit of trial and error to get Sphinx-lint to accept the new text, but I think it’s finally in good shape.